File: DataStructure_Alogorithm/Algorithm/Inflearn_basic_cpp/23_연속부분증가수열/main.cpp
#include<stdio.h>
#include<vector>

// 방법 1 . 이전 값과 비교 해서 크면 +를 해준다.
// max에 넣어주고 max랑 비교를 해준다.

int main() {
	int n, i, pre, now, cnt, max;
	scanf("%d", &n);
	scanf("%d", &pre);
	cnt = 1;
	max = 1;

	for(i=2; i<=n; i++) {
		scanf("%d", &now);
		if(now >= pre) {
			cnt++;
			if (cnt > max) {
				max = cnt;
			}
		} else cnt=1;
		pre=now;
	}
	
	printf("%d\n", max);


	return 0;
}

File: DataStructure_Alogorithm/Algorithm/Inflearn_basic_cpp/23_연속부분증가수열/main.go
package main

import (
	"fmt"
	"os"
)

func main() {
	file, _ := os.Open("input.txt")
	defer file.Close()
	old := os.Stdin
	defer func() { os.Stdin = old }()
	os.Stdin = file

	// 연속적으로 중가하는 수열의 최대 길이
	// 값이 같을 땐 증가하는 것으로 간주

	var N int
	fmt.Scanf("%d", &N)
	arr := make([]int, N)
	for i := 0; i < N; i++ {
		fmt.Scanf("%d", &arr[i])
	}
	cnt := 1
	max := 1
	for i := 1; i < N; i++ {
		if arr[i-1] <= arr[i] {
			cnt++
		} else {
			cnt = 1
		}

		if cnt > max {
			max = cnt
		}
	}

	fmt.Println(max)
}
